contrib/inventory/hyperone.py

```diff
@@ -0,0 +1,215 @@
+
+        for tag in vm.get('tag', {}).keys() or [self.default_group]:
+            self.push(self.inventory, tag, name)
+
+        self.push(self.inventory, 'hyperone', name)
+
+        hostvars = {'hyperone': vm}
+
+        if vm['private_ips']:
+            hostvars['private_ip'] = vm['private_ips'][0]
+
+        if self.connection_host:
+            if vm['private_ips'] and self.connection_host == 'private_ip':
+                hostvars['ansible_ssh_host'] = vm["private_ips"][0]
+                hostvars['ansible_host'] = vm["private_ips"][0]
+
+            if vm['public_ips'] and (self.connection_host == 'public_ip' or not vm['private_ips']):
+                hostvars['ansible_ssh_host'] = vm["public_ips"][0]
+                hostvars['ansible_host'] = vm["public_ips"][0]
+
+        if self.update_username:
+            hostvars['ansible_ssh_user'] = vm.get('data', {}).get('username', 'guru')
+
+        self.inventory["_meta"]["hostvars"][name] = hostvars
+
+    def _get_ips(self, netadps, type='public'):
+        return [ip['address']
+                for netadp in netadps if netadp['network']['type'] == type
+                for ip in netadp['ip']]
+
+    def get_inventory(self):
+        if self.cache.is_valid():
+            return self.cache.get()
+
+        vms = self.api.get_vms()
+        for vm in vms:
+            vm['netadps'] = self.api.get_vm_netadp(vm_id=vm['_id'])
+            vm['public_ips'] = self._get_ips(vm['netadps'])
+            vm['private_ips'] = self._get_ips(vm['netadps'], 'private')
+            vm['ips'] = vm['private_ips'] + vm['public_ips']
+            self.add_vm(vm)
+        self.cache.save(self.inventory)
+        return self.inventory
+
+    def __init__(self):
+
+        self.token = None
+        self.connection_host = None
+        self.default_group = 'hyperone_untagged'
+        self.cache_path_inventory = None
+        self.cache_max_age = 60 * 60
+        self.update_username = False
+        self.read_settings()
+        self.read_env()
+        self.cache = FileCache(self.cache_path_inventory, self.cache_max_age)
+        self.inventory = {'_meta': {'hostvars': {}}}
+
+        options = self.parse_cli_args()
+
+        if self.token:
+            self.api = HyperOneAPI(self.token)
+
+            if options.list:
+                print(json.dumps(self.get_inventory(), indent=2))
+
+            elif options.host:
+                host = self.get_inventory()['_meta']['hostvars'][options.host]
+                print(json.dumps(host, indent=2))
+
+            elif options.purge_cache:
+                os.unlink(self.cache_path_inventory)
+                print("Cache purged!")
+
+            else:
+                print("usage: --list  ..OR.. --host <hostname> ..OR.. --purge-cache", file=sys.stderr)
+                sys.exit(1)
+
+        else:
+            print("Error: Configuration of toekn are required. See hyperone.ini.")
+            sys.exit(1)
+
+
+HyperOneInventory()
```

Comment:
This should probably guarded with `if __name__ == '__main__':`

Reply:
I follow style of other script. I unifed them in https://github.com/ansible/ansible/pull/44453

```
$ diff <(grep '' contrib/inventory/ -l -R) <(grep '__main__' contrib/inventory/ -l -R) | grep '.py' | wc -l
25
```